lib/bamboo/mailer.ex

```diff
@@ -142,6 +142,7 @@ defmodule Bamboo.Mailer do
       Email.welcome_email
       |> Mailer.deliver_now(config: %{username: "Emma", smtp_port: 2525})
   """
+  @spec deliver_now(any()) :: no_return()
```

Comment:
What do you think about changing the `any()` to `Email.t()` in all of these `@spec` definition changes? I think the first argument is supposed to be an email (even though I realize these functions only `raise` an error).